entity-server-api/src/main/java/org/terracotta/entity/ServiceRegistry.java

```diff
@@ -16,24 +16,38 @@
  *  Terracotta, Inc., a Software AG company
  *
  */
-
 package org.terracotta.entity;
 
+
 /**
- * Registry which Entities interact with list of services provided by the platform. Services provided to entity might
- * be namespaced or not, depending on  service provider.
- *
- * @author twu
+ * The registry given to entities so that they can look up the services hosted on the platform.
+ * 
+ * Note that the Service instances returned by this interface are not cached or interned, in any way (unless the underlying
+ *  ServiceProvider decides to add that functionality).  This means that multiple requests for the same service, through the
+ *  same registry, will return different instances.
+ * 
+ * ServiceProvider implementations see each call through a ServiceRegistry as originating from a specific consumerID, which
```

Comment:
consumerIDs are described in greater detail in ServiceProvider so I mentioned that as a link in this update.